File: ThuongMaiNongSanApp/lib/models/register_model.dart

class RegisterModel {
  final String userName;
  final String tenKhachHang;
  final String diaChi;
  final String email;
  final String password;
  final String role;

  const RegisterModel({
    required this.userName,
    required this.tenKhachHang,
    required this.diaChi,
    required this.email,
    required this.password,
    required this.role
  });

  Map<String, dynamic> toJson() {
    return {
      "userName": userName,
      "tenKhachHang": tenKhachHang,
      "diaChi": diaChi,
      "role": role,
      "email": email,
      "password": password,
    };
  }
}

File: ThuongMaiNongSanWeb/MobileApp/DTO/RegisterModel.cs
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations;

namespace WebApp.MobileApp.DTO
{
    public class RegisterModel
    {
        [Required]
        public string Username { get; set; } = null!;
        [Required]
        public string TenKhachHang { get; set; } = null!;
        [Required]
        public string DiaChi { get; set; } = null!;
        [Required]
        public IFormFile Avatar { get; set; } = null!;
        [Required]
        public string Email { get; set; } = null!;
        [Required]
        public string Password { get; set; } = null!;
        [Required]
        public string Role { get; set; } = null!;
    }
}
